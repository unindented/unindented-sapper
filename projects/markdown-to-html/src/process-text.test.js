const dictionary = require("dictionary-en-us");
const spell = require("retext-spell");
const stringify = require("retext-stringify");
const processor = require("./process-text");

const process = text =>
  new Promise((resolve, reject) => {
    processor()
      .use(spell, dictionary)
      .use(stringify)
      .process(text, (err, file) => {
        if (!err) {
          resolve(file.messages);
        } else {
          reject(err);
        }
      });
  });

describe("text", () => {
  it("leaves mentions alone", () => {
    const text = `
Hello @unindented!
`;
    return expect(process(text)).resolves.toMatchSnapshot();
  });

  it("leaves urls alone", () => {
    const text = `
Visit me at unindented.org!
`;
    return expect(process(text)).resolves.toMatchSnapshot();
  });

  it("warns about insensitive, inconsiderate language", () => {
    const text = `
For he is truly his brother’s keeper, and the finder of lost children.
`;
    return expect(process(text)).resolves.toMatchSnapshot();
  });

  it("warns about passive voice", () => {
    const text = `
I was led to believe that if they come home and find us here, they wouldn’t appreciate it none too much?
`;
    return expect(process(text)).resolves.toMatchSnapshot();
  });

  it("warns about profane and vulgar wording", () => {
    const text = `
Why do we feel it’s necessary to yak about bullshit in order to be comfortable?
`;
    return expect(process(text)).resolves.toMatchSnapshot();
  });

  it("warns about apostrophes in elided contractions", () => {
    const text = `
In the past six years, 50’s diners have sprung up all over L.A.
`;
    return expect(process(text)).resolves.toMatchSnapshot();
  });

  it("warns about use of diacritics", () => {
    const text = `
Decor out of an “Archie” comic book.
`;
    return expect(process(text)).resolves.toMatchSnapshot();
  });

  it("warns about indefinite articles", () => {
    const text = `
Was than a uncomfortable silence?
`;
    return expect(process(text)).resolves.toMatchSnapshot();
  });

  it("warns about quotes", () => {
    const text = `
You still wanna hear my "Fox Force Five" joke?
`;
    return expect(process(text)).resolves.toMatchSnapshot();
  });

  it("warns about redundant acronyms", () => {
    const text = `
They stand nervously at an ATM machine.
`;
    return expect(process(text)).resolves.toMatchSnapshot();
  });

  it("warns about repeated words", () => {
    const text = `
How long do you intend to to walk the earth?
`;
    return expect(process(text)).resolves.toMatchSnapshot();
  });

  it("warns about spacing between sentences", () => {
    const text = `
Now I wanna dance, I wanna win.  I want that trophy, so dance good.
`;
    return expect(process(text)).resolves.toMatchSnapshot();
  });
});
