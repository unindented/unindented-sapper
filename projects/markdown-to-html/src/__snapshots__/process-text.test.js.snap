// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`text leaves mentions alone 1`] = `Array []`;

exports[`text leaves urls alone 1`] = `Array []`;

exports[`text warns about apostrophes in elided contractions 1`] = `
Array [
  [2:24-2:28: Expected the apostrophe in \`50’s\` to be like this: \`’50s\`],
]
`;

exports[`text warns about indefinite articles 1`] = `
Array [
  [2:10-2:11: Use \`an\` before \`uncomfortable\`, not \`a\`],
]
`;

exports[`text warns about insensitive, inconsiderate language 1`] = `
Array [
  [2:17-2:20: \`his\` may be insensitive, when referring to a person, use \`their\`, \`theirs\`, \`them\` instead],
  [2:21-2:30: \`brother’s\` may be insensitive, use \`siblings\` instead],
  [2:5-2:7: \`he\` may be insensitive, use \`they\`, \`it\` instead],
]
`;

exports[`text warns about passive voice 1`] = `
Array [
  [2:7-2:10: Don’t use the passive voice],
]
`;

exports[`text warns about profane and vulgar wording 1`] = `
Array [
  [2:44-2:52: Don’t use “bullshit”, it’s profane],
]
`;

exports[`text warns about quotes 1`] = `
Array [
  [2:25-2:26: Expected a smart quote: \`“\`, not \`"\`],
  [2:40-2:41: Expected a smart quote: \`”\`, not \`"\`],
]
`;

exports[`text warns about redundant acronyms 1`] = `
Array [
  [2:28-2:39: Replace \`ATM machine\` with \`ATM\`],
]
`;

exports[`text warns about repeated words 1`] = `
Array [
  [2:24-2:29: Expected \`to\` once, not twice],
]
`;

exports[`text warns about spacing between sentences 1`] = `
Array [
  [2:32-2:34: Expected \`1\` space between sentences, not \`2\`],
]
`;

exports[`text warns about use of diacritics 1`] = `
Array [
  [2:1-2:6: Replace \`Decor\` with \`Décor\`],
]
`;
